fotoapparat/src/main/java/io/fotoapparat/parameter/range/DiscreteRange.java

```diff
@@ -0,0 +1,68 @@
+package io.fotoapparat.parameter.range;
+
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import io.fotoapparat.util.StringUtils;
+
+/**
+ * Implementation of {@link Range} that represents set of given values.
+ *
+ * @param <T> type of objects in that set.
+ */
+class DiscreteRange<T extends Comparable<? super T>> implements Range<T> {
+    @NonNull private final List<T> values;
+
+    public DiscreteRange(@NonNull Collection<T> values) {
+        List<T> valuesList = (values instanceof List)
+                ? (List<T>) values
+                : new ArrayList<>(values);
+        Collections.sort(valuesList);
+        this.values = valuesList;
+    }
+
+    @Override
+    public boolean contains(T value) {
+        return values.contains(value);
+    }
+
+    @Override
+    public T highest() {
```

Comment:
Makes sense to add `@Nullable` annotation.

Reply:
Sure.